pkg/suggestion/nasrl_service.py

```diff
@@ -18,9 +17,21 @@ def __init__(self, logger=None):
         self.manager_port = 6789
         self.registered_studies = list()
 
-        self.ctrl_cache_file = ""
-        self.ctrl_step = 0
-        self.is_first_run = True
+        self.study_names = dict()
+        self.tf_graphs = dict()
+        self.prev_trial_id = dict()
+        self.ctrl_cache_file = dict()
+        self.ctrl_step = dict()
+        self.is_first_run = dict()
+        self.suggestion_configs = dict()
+        self.controllers = dict()
+        self.num_layers = dict()
+        self.input_size = dict()
+        self.output_size = dict()
+        self.num_operations = dict()
+        self.search_space = dict()
+        self.opt_direction = dict()
+        self.objective_name = dict()
```

Comment:
how about self.studyinfo = {"studyid": {name: "", "tf_graph": xx, "prev_trial_id": xx, "ctrl_cache_file": xx, ..., "objective_name": xx}} to replace above dicts since they has same key?

Reply:
I have reorganized the program a little bit. Now, the information of every StudyJob is stored in a `NAS_RL_StudyJob` class. And the initialization of a StudyJob and retrieving parameters are all handled by this class. In this way, GetSuggestion gets much cleaner.